cdap-storage-ext-spanner/src/main/java/io/cdap/cdap/storage/spanner/SpannerStructuredTable.java

```
@@ -71,6 +71,10 @@ public class SpannerStructuredTable implements StructuredTable {
    * Spanner limit for byte[] field in bytes.
    */
   private static final int BYTES_LIMIT = 10 * 1024 * 1024;
+  /**
+   * Pipeline limit (in MB) displayed to the user if field size exceeds spanner limit.
+   */
+  private static final int PIPELINE_LIMIT_MB = 2;
```

why do we need to hardcode this? If tomorrow spanner changes the limit?

Spanner must be throwing a specific exception for this right?
 
Can't we catch it and then throw a descriptive error message?

---

Spanner throws error message like : 
```
com.google.cloud.spanner.SpannerException: FAILED_PRECONDITION: com.google.api.gax.rpc.FailedPreconditionException: io.grpc.StatusRuntimeException: FAILED_PRECONDITION: New value exceeds the maximum size limit for this column in this database: application_specs.application_data, size: 2860698, limit: 2621440.
```

We cannot propagate this to the users. 

This is kind of a temporary solution until we have chunking in place.

---

I didn't mean to display spanner message to users but rather catch this exception and throw the error after checking that spanner emitted it

---

Yes right, but there are 2 blockers in that approach :
1. error is returned with FAILED_PRECONDITION code. We cannot depend on the error code to identify if it is the cell size issue.
2. Currently string compression is dependent on the STRING_LIMIT constant. We would also do chunking in future based on this attribute.
If in future spanner decides to increase their limits, we would for sure need cdap changes to increase those limits.

Hence, we are trying to fail fast by doing the validation even before we perform any spanner operations.